lib/nimble.phx.gen.template/addons/testing_tool.ex

```diff
@@ -0,0 +1,160 @@
+defmodule Nimble.Phx.Gen.Template.Addons.TestingTool do
+  use Nimble.Phx.Gen.Template.Addon
+
+  @versions %{
+    credo: "1.4",
+    dialyxir: "1.0",
+    ex_machina: "2.4",
+    mox: "0.5.2",
+    excoveralls: "0.12.2"
+  }
+
+  @impl true
+  def copy_files(%Project{} = project) do
+    binding = [
+      otp_app: project.otp_app,
+      base_module: project.base_module,
+      minimum_coverage: 100
+    ]
+
+    files = [
+      {:text, ".credo.exs", ".credo.exs"},
+      {:eex, "coveralls.json.eex", "coveralls.json"},
+      {:eex, "test/support/factory.ex.eex", "test/support/factory.ex"},
+      {:eex, "test/support/mock.ex.eex", "test/support/mock.ex"}
+    ]
+
+    Generator.copy_file(files, binding)
+
+    project
+  end
+
+  @impl true
+  def edit_files(%Project{} = project) do
+    project
+    |> inject_mix_dependency
+    |> edit_mix
+    |> edit_test_config
+    |> edit_test_helper
+    |> import_factory
+
+    project
+  end
+
+  defp inject_mix_dependency(project) do
+    [
+      {:credo, "~> " <> @versions.credo, only: [:dev, :test], runtime: false},
+      {:dialyxir, "~> " <> @versions.dialyxir, only: [:dev], runtime: false},
+      {:excoveralls, "~> " <> @versions.excoveralls, only: :test},
+      {:ex_machina, "~> " <> @versions.ex_machina, only: :test},
+      {:mox, "~> " <> @versions.mox, only: :test}
```

Comment:
Right now, I merge all the testing stuff into only 1 file the `testing_tool.ex` and I fell that too much content in that module, I have an idea to separate each addon like credo addon, wallaby addon, sobelow addon, … What do you think?